src/PKSim.Core/Snapshots/Mappers/OriginDataMapper.cs

```diff
@@ -49,18 +50,34 @@ ISpeciesRepository speciesRepository
             x.Species = originData.Species.Name;
             x.Population = originData.Species.Populations.Count > 1 ? originData.SpeciesPopulation.Name : null;
             x.Gender = originData.SpeciesPopulation.Genders.Count > 1 ? originData.Gender.Name : null;
-            x.Age = parameterFrom(originData.Age, originData.AgeUnit, _dimensionRepository.AgeInYears, originData.SpeciesPopulation.IsAgeDependent);
-            x.GestationalAge = parameterFrom(originData.GestationalAge, originData.GestationalAgeUnit, _dimensionRepository.AgeInWeeks, originData.SpeciesPopulation.IsAgeDependent);
-            x.Height = parameterFrom(originData.Height, originData.HeightUnit, _dimensionRepository.Length, originData.SpeciesPopulation.IsHeightDependent);
-            x.Weight = parameterFrom(originData.Weight, originData.WeightUnit, _dimensionRepository.Mass);
-
          });
 
+         if (originData.SpeciesPopulation.IsAgeDependent)
+         {
+            snapshot.Age = originDataParameterFor(originData, _individualModelTask.MeanAgeFor, originData.Age, originData.AgeUnit, _dimensionRepository.AgeInYears);
+            snapshot.GestationalAge = originDataParameterFor(originData, _individualModelTask.MeanGestationalAgeFor, originData.GestationalAge, originData.GestationalAgeUnit, _dimensionRepository.AgeInWeeks);
+         }
+
+         snapshot.Weight = originDataParameterFor(originData, _individualModelTask.MeanWeightFor, originData.Weight, originData.WeightUnit, _dimensionRepository.Mass);
+
+         if (originData.SpeciesPopulation.IsHeightDependent)
+            snapshot.Height = originDataParameterFor(originData, _individualModelTask.MeanHeightFor, originData.Height, originData.HeightUnit, _dimensionRepository.Length);
+
          snapshot.ValueOrigin = await _valueOriginMapper.MapToSnapshot(originData.ValueOrigin);
          snapshot.CalculationMethods = await _calculationMethodCacheMapper.MapToSnapshot(originData.CalculationMethodCache, originData.Species.Name);
          return snapshot;
       }
 
+      private Parameter originDataParameterFor(ModelOriginData originData, Func<ModelOriginData, IParameter> meanParameterRetrieverFunc, double? value, string unit, IDimension dimension)
+      {
+         var meanParameter = meanParameterRetrieverFunc(originData);
+
+         if (value == null || ValueComparer.AreValuesEqual(meanParameter.Value, value.Value))
```

Comment:
Check if the value of any of those parameters is equal to the mean value for the origin data

MeanAge = func(Species, Pop, Gender)
MeanGA =  func(Species, Pop, Gender)
MeanWeight = func(species, pop, gender, Age, GA)
MeanHeight = func(species, pop, gender, Age, GA)